### DISCUSSION TITLE
Help adding vscode snippets from package

### DISCUSSION THREAD

**Discussion by gugahoi** [L3MON4D3/LuaSnip#939] [Q&A] [answered]
I am trying to get some snippets from [vscode-jest-snippets](https://github.com/andys8/vscode-jest-snippets) working in NeoVim but no matter what I try to do, they don't seem to get recognized. Can anyone point me in the right direction? Below are my configs (stripped down for brevity).

For reference, snippets from `friendly-snippets` are working correctly so I am unsure if `vscode-jest-snippets` are just incompatible or if I am doing something incorrectly.

Hoping someone can help me.
Thanks in advance!

<details>

<summary>luasnip.lua</summary>

```lua
return {
	"L3MON4D3/LuaSnip",
	dependencies = {
		{
			"rafamadriz/friendly-snippets",
			config = function()
				require("luasnip.loaders.from_vscode").lazy_load()
			end,
		},
		{
			"andys8/vscode-jest-snippets",
			config = function()
				require("luasnip.loaders.from_vscode").lazy_load()
			end,
		},
	},
	config = function()
		require("luasnip").filetype_extend("typescript", { "javascript" })
	end,
	keys = {
		{
			"<tab>",
			function()
				return require("luasnip").jumpable(1) and "<Plug>luasnip-jump-next" or "<tab>"
			end,
			expr = true,
			silent = true,
			mode = "i",
		},
		{
			"<tab>",
			function()
				require("luasnip").jump(1)
			end,
			mode = "s",
		},
		{
			"<s-tab>",
			function()
				require("luasnip").jump(-1)
			end,
			mode = { "i", "s" },
		},
	},
}
```

</details> 

<details>

<summary>cmp.lua</summary>

```lua
return {
	"hrsh7th/nvim-cmp",
	dependencies = {
		"hrsh7th/cmp-nvim-lsp",
		"hrsh7th/cmp-buffer",
		"hrsh7th/cmp-cmdline",
		"petertriho/cmp-git",
		"saadparwaiz1/cmp_luasnip",
		"rafamadriz/friendly-snippets",
		"L3MON4D3/LuaSnip",
	},
	config = function()
		-- luasnip setup
		local luasnip = require("luasnip")
		local cmp = require("cmp")

		local has_words_before = function()
			local line, col = unpack(vim.api.nvim_win_get_cursor(0))
			return col ~= 0 and vim.api.nvim_buf_get_lines(0, line - 1, line, true)[1]:sub(col, col):match("%s") == nil
		end

		local compare = require("cmp.config.compare")

		-- nvim-cmp setup
		cmp.setup({
			snippet = {
				expand = function(args)
					luasnip.lsp_expand(args.body)
				end,
			},
			mapping = cmp.mapping.preset.insert({
				["<C-b>"] = cmp.mapping.scroll_docs(-4),
				["<C-f>"] = cmp.mapping.scroll_docs(4),
				["<C-Space>"] = cmp.mapping.complete(),
				["<C-e>"] = cmp.mapping.abort(),
				["<CR>"] = cmp.mapping.confirm({
					select = true,
				}),
				["<Tab>"] = cmp.mapping(function(fallback)
					if cmp.visible() then
						cmp.select_next_item()
					elseif luasnip.expand_or_jumpable() then
						luasnip.expand_or_jump()
					elseif has_words_before() then
						cmp.complete()
					else
						fallback()
					end
				end, { "i", "s" }),

				["<S-Tab>"] = cmp.mapping(function(fallback)
					if cmp.visible() then
						cmp.select_prev_item()
					elseif luasnip.jumpable(-1) then
						luasnip.jump(-1)
					else
						fallback()
					end
				end, { "i", "s" }),
			}),
			sources = cmp.config.sources({
				{ name = "luasnip" },
			}),
		})
	end,
}
```

</details>

**Comment by L3MON4D3** [comment]
Could you post the logfile? `:lua require("luasnip").log.open()` will open the file.
Best  clear it, and then restart neovim so there isn't too much noise

**Reply by gugahoi** [reply]
I deleted the contents and restarted Neovim and here are the logs:

```
WARN  | vscode-loader: Package /Users/guga/Projects/gugahoi/yarn.nvim/package.json does not contribute any snippets, skipping it
WARN  | vscode-loader: Package /Users/guga/.local/share/nvim/lazy/typescript.nvim/package.json does not contribute any snippets, skipping it
```

**Reply by gugahoi** [reply]
I've managed to get it to work now doing the following:

1. Clone https://github.com/andys8/vscode-jest-snippets
2. Add the following to the setup function for LuaSnip
```lua
require("luasnip.loaders.from_vscode").lazy_load({
  paths = "/Users/guga/Projects/andys8/vscode-jest-snippets",
})
```
3. Remove the "source" fields in all snippets in `vscode-jest-snippets` `snippets.json` file

Is it possible the `source` field is causing compatibility problems?

**Reply by L3MON4D3** [reply, marked as answer]
OH it's possible they are just in the "wrong" filetype, with`require("luasnip").filetype_extend("javascript", {"js"})`, they should show up correctly, even with scope set

**Reply by gugahoi** [reply]
Perfect, that did it! Out of curiosity, is there a reason why `js` and `javascript` would be separate and why would either of them not also be included in `typescript` snippets?

**Reply by L3MON4D3** [reply]
Stuff like that is just not set up by default :D
Trying to do something smart might backfire, and then I'd prefer that it's just easy to set up yourself

**Reply by gugahoi** [reply]
I see, it makes sense. However now I am finding several snippets that should be enabled for my languages that are not, for example: `jsdoc` and `tsdoc`. There doesn't seem to be an easy way to look into what is/should be available other then investigating the source code.

In any case thank you for your help and the tools you've developed, it's much appreciated!
